src/messageComposer/middleware/textComposer/applyCommandSettings.ts

```diff
@@ -0,0 +1,55 @@
+import type { Middleware } from '../../../middleware';
+import type { TextComposerMiddlewareExecutorState } from './TextComposerMiddlewareExecutor';
+import type { CommandSuggestion } from './types';
+
+export type ApplyCommandSettingsMiddleware = Middleware<
+  TextComposerMiddlewareExecutorState<CommandSuggestion>,
+  'onChange' | 'onSuggestionItemSelect'
+>;
+
+export const createApplyCommandSettingsMiddleware =
+  (): ApplyCommandSettingsMiddleware => ({
+    handlers: {
+      onChange: ({ complete, forward, state }) => {
+        const { command } = state;
+
+        if (!command) {
+          return forward();
+        }
+
+        const trigger = `/${command.name}`;
+        const newText = state.text.replace(new RegExp(`^${trigger}(\\s|$)`), '');
+
+        return complete({
+          ...state,
+          selection: {
+            end: state.selection.end - trigger.length,
+            start: state.selection.start - trigger.length,
+          },
+          suggestions: undefined,
+          text: newText,
+        });
+      },
+      onSuggestionItemSelect: ({ complete, forward, state }) => {
+        const { command } = state;
+
+        if (!command) {
+          return forward();
+        }
+
+        const trigger = `/${command?.name} `;
+
+        const newText = state.text.slice(trigger.length);
+        return complete({
+          ...state,
+          selection: {
+            end: state.selection.end - trigger.length,
+            start: state.selection.start - trigger.length,
+          },
+          suggestions: undefined,
```

Comment:
Why are the suggestions set to undefined here? I would expect this middleware to take care only of the text.

Reply:
I can still see it : )

Reply:
Ah, I missed pushing it.